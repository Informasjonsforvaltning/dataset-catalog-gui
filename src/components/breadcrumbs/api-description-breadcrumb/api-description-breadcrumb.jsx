import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import _ from 'lodash';

import { getApiItemsByApiId } from '../../../redux/modules/apis';

export const PureapiDescriptionBreadcrumb = props => {
  const { apiDescriptionItem } = props;
  return (
    <span>
      {_.get(apiDescriptionItem, ['apiSpecification', 'info', 'title'])}
    </span>
  );
};

const mapStateToProps = ({ apis }, ownProps) => {
  const catalogId = _.get(ownProps, ['match', 'params', 'catalogId']);
  const apiId = _.get(ownProps, ['match', 'params', 'id']);
  return {
    apiDescriptionItem: getApiItemsByApiId(apis, catalogId, apiId)
  };
};

PureapiDescriptionBreadcrumb.defaultProps = {
  apiDescriptionItem: null
};

PureapiDescriptionBreadcrumb.propTypes = {
  apiDescriptionItem: PropTypes.object
};

export const ApiDescriptionBreadcrumb = connect(mapStateToProps)(
  PureapiDescriptionBreadcrumb
);
