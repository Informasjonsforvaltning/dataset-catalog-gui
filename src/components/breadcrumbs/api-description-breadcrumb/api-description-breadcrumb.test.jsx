import React from 'react';
import { shallow } from 'enzyme';
import { PureapiDescriptionBreadcrumb } from './api-description-breadcrumb';

test('should render Pureapi-descriptionBreadcrumb with "api-description" active correctly', () => {
  const defaultProps = {
    apiDescriptionItem: {
      title: 'api-description title'
    }
  };
  const wrapper = shallow(<PureapiDescriptionBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Pureapi-descriptionBreadcrumb correctly with no api-descriptionItem', () => {
  const defaultProps = {};
  const wrapper = shallow(<PureapiDescriptionBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
