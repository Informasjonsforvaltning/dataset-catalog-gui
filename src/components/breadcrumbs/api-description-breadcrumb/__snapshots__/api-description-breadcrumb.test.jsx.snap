// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `should render Pureapi-descriptionBreadcrumb correctly with no api-descriptionItem 1`
] = `<span />`;

exports[
  `should render Pureapi-descriptionBreadcrumb with "api-description" active correctly 1`
] = `<span />`;
